lib/JSONSchemaGenerator.js

```diff
@@ -60,10 +61,33 @@ const {
 } = require('./consts')
 
 const EXAMPLE_DATE = new Date('2020-09-16T12:00:00Z').toISOString()
-
 const mandatoryFields = new Set([MONGOID, UPDATERID, UPDATEDAT, CREATORID, CREATEDAT, __STATE__])
-
-const typesToIgnoreInQuerystring = new Set([ARRAY, GEOPOINT, RAWOBJECTTYPE])
+const mandatoryFieldsWithoutId = (isNewSchema) => ({
+  [UPDATERID]: {
+    type: 'string',
+    description: 'User id that has requested the last change successfully',
+  },
+  [UPDATEDAT]: {
+    type: 'string',
+    ...isNewSchema ? { format: 'date-time' } : { pattern: '^\\d{4}-\\d{2}-\\d{2}(T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,3})?(Z|[+-]\\d{2}:\\d{2}))?$' },
+    description: 'Date of the request that has performed the last change',
+    examples: [
+      '2020-09-16T12:00:00.000Z',
+    ],
+  },
+  [CREATORID]: {
+    type: 'string',
+    description: 'User id that has created this object',
+  },
+  [CREATEDAT]: {
+    type: 'string',
+    ...isNewSchema ? { format: 'date-time' } : { pattern: '^\\d{4}-\\d{2}-\\d{2}(T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,3})?(Z|[+-]\\d{2}:\\d{2}))?$' },
+    description: 'Date of the request that has performed the object creation',
+    examples: [
+      '2020-09-16T12:00:00.000Z',
+    ],
+  },
```

Comment:
I'm a bit obnoxious here, sorry.

```suggestion
  [CREATORID]: {
    type: 'string',
    description: 'User id that has created this object',
  },
  [CREATEDAT]: {
    type: 'string',
    ...isNewSchema ? { format: 'date-time' } : { pattern: '^\\d{4}-\\d{2}-\\d{2}(T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,3})?(Z|[+-]\\d{2}:\\d{2}))?$' },
    description: 'Date of the request that has performed the object creation',
    examples: [
      '2020-09-16T12:00:00.000Z',
    ],
  },
  [UPDATERID]: {
    type: 'string',
    description: 'User id that has requested the last change successfully',
  },
  [UPDATEDAT]: {
    type: 'string',
    ...isNewSchema ? { format: 'date-time' } : { pattern: '^\\d{4}-\\d{2}-\\d{2}(T\\d{2}:\\d{2}:\\d{2}(\\.\\d{1,3})?(Z|[+-]\\d{2}:\\d{2}))?$' },
    description: 'Date of the request that has performed the last change',
    examples: [
      '2020-09-16T12:00:00.000Z',
    ],
  },
```